crates/orchestrator/src/jobs/state_update_job/mod.rs

```diff
@@ -31,26 +53,207 @@ impl Job for StateUpdateJob {
         })
     }
 
-    async fn process_job(&self, _config: &Config, _job: &JobItem) -> Result<String> {
+    async fn process_job(&self, config: &Config, job: &mut JobItem) -> Result<String> {
+        let attempt_no =
+            job.metadata.get(JOB_PROCESS_ATTEMPT_METADATA_KEY).expect("Could not find current attempt number.").clone();
+
+        // TODO: remove when SNOS is correctly stored in DB/S3
+        // Test metadata to fetch the snos output from the test folder, default to False
+        let fetch_from_tests =
+            job.metadata.get(JOB_METADATA_STATE_UPDATE_FETCH_FROM_TESTS).map_or(false, |value| value == "TRUE");
+
         // Read the metadata to get the blocks for which state update will be performed.
-        // For each block, get the program output (from the PIE?) and the
-        todo!()
+        // We assume that blocks nbrs are formatted as follow: "2,3,4,5,6".
+        let block_numbers = self.get_block_numbers_from_metadata(job)?;
+        self.validate_block_numbers(config, &block_numbers).await?;
```

Comment:
we do need skip blocks which have already been processed in case this is a retry (we need to check `JOB_METADATA_STATE_UPDATE_LAST_FAILED_BLOCK_NO` key)

Reply:
Done